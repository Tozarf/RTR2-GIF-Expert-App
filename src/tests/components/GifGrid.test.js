import React from 'react'
import "@testing-library/jest-dom"
import { GifGrid } from '../../components/GifGrid'
import { shallow } from 'enzyme'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')

describe('Testing <GifGrid/>', () => {
    const category = "the simpsons"
    // const wrapper = shallow(<GifGrid category={category}/>)
    test('should match snapshot', () => {
        
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        
        const wrapper = shallow(<GifGrid category={category}/>)

        expect(wrapper).toMatchSnapshot();
    })
    test('should show items when images are loaded (useFetchGifs)', () => {
        
        const gifs= [{
            id: "ABC",
            url:"https://localhost/cualquiercosa.jpg",
            title: "cualquier cosa"
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow(<GifGrid category={category}/>)
        // console.log(wrapper.find("GifGridItem"))
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find("p").exists()).toBe(false)
        expect(wrapper.find("GifGridItem").length).toBe(gifs.length)
    })

    
    
    

})
