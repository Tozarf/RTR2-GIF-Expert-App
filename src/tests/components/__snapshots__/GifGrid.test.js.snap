// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifGrid/> should match snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    the simpsons
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Testing <GifGrid/> should show items when images are loaded (useFetchGifs) 1`] = `
<Fragment>
  <h3
    className="animate__animated animate_fadeIn"
  >
    the simpsons
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="cualquier cosa"
      url="https://localhost/cualquiercosa.jpg"
    />
  </div>
</Fragment>
`;
